src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,502 @@
+            try:
+                send_single_erasure_email(
+                    db=db,
+                    subject_email=batched_identity_data.email_address,
+                    subject_name=batched_identity_data.vendor_name,
+                    batch_identities=batched_identity_data.identities,
+                    test_mode=False,
+                )
+            except MessageDispatchException as exc:
+                logger.info("Erasure email failed with exception {}", exc)
+                raise
+
+        # create an audit event for each privacy request ID
+        for privacy_request in privacy_requests:
+            if privacy_request.id not in skipped_privacy_requests:
+                ExecutionLog.create(
+                    db=db,
+                    data={
+                        "connection_key": self.configuration.key,
+                        "dataset_name": self.configuration.name
+                        or self.configuration.key,
+                        "collection_name": self.configuration.name
+                        or self.configuration.key,
+                        "privacy_request_id": privacy_request.id,
+                        "action_type": ActionType.erasure,
+                        "status": ExecutionLogStatus.complete,
+                        "message": f"Erasure email instructions dispatched for '{self.configuration.name}'",
+                    },
+                )
+
+    def get_collection_and_field_from_reference(
+        self,
+        db: Session,
+        privacy_requests: Query,
+        dataset_reference: FidesDatasetReference,
+    ) -> Tuple[str, str]:
+        dataset_key = dataset_reference.dataset  # pylint: disable=no-member
+
+        # Field should be of the form collection_name.field_name, potentially with nested fields
+        # e.g collection_name.field_name.nested_field_name, so splitting by "." should give
+        # a list of length greater than or equal to 2.
+        split_field = dataset_reference.field.split(".")  # pylint: disable=no-member
+
+        if len(split_field) < 2:
+            field = dataset_reference.field  # pylint: disable=no-member
+            error_log = f"Invalid dataset reference field {field} for dataset {dataset_key}. Skipping erasure email send for connector: {self.configuration.key}."
+            logger.error(error_log)
+            self.error_all_privacy_requests(
+                db,
+                privacy_requests,
+                f"Connector configuration references invalid dataset field {field} for dataset {dataset_key}.",
+            )
+            raise DynamicErasureEmailConnectorException(error_log)
+
+        # Get collection name and field name
+        collection_name = split_field[0]
+        field_name = ".".join(split_field[1:])
+
+        return collection_name, field_name
+
+    def process_connector_config(
+        self, db: Session, privacy_requests: Query
+    ) -> ProcessedConfig:
```

Comment:
The function name/ProcessedConfig is throwing me here - maybe it's the word "config". which has other usages in our codebase..

Reply:
we'll just go with this, the excellent docstrings you've added to this file help explain some of the non-standard operations you're doing